src/object/cli_obj.c

```diff
@@ -493,6 +554,144 @@ obj_shard2tgtid(struct dc_object *obj, uint32_t shard)
 	return obj->cob_shards->do_shards[shard].do_target_id;
 }
 
+/**
+ * Create reasb_req and set iod'svalue, akey reuse buffer from input
+ * iod, iod_type/iod_size assign as input iod, iod_kcsum/iod_nr/iod_recx/
+ * iod_csums/iod_eprs array will set as 0/NULL.
+ */
+static int
+obj_reasb_req_init(struct obj_auxi_args *obj_auxi, daos_iod_t *iods,
+		   uint32_t iod_nr, struct daos_oclass_attr *oca)
+{
+	struct obj_reasb_req		*reasb_req = &obj_auxi->reasb_req;
+	daos_size_t			 size_iod, size_sgl, size_oiod;
+	daos_size_t			 size_recx, size_tgt_nr;
+	daos_size_t			 size_sorter, buf_size;
+	daos_iod_t			*uiod, *riod;
+	struct obj_ec_recx_array	*ec_recx;
+	void				*buf;
+	uint8_t				*tmp_ptr;
+	int				 i;
+
+	size_iod = roundup(sizeof(daos_iod_t) * iod_nr, 8);
+	size_sgl = roundup(sizeof(d_sg_list_t) * iod_nr, 8);
+	size_oiod = roundup(sizeof(struct obj_io_desc) * iod_nr, 8);
+	size_recx = roundup(sizeof(struct obj_ec_recx_array) * iod_nr, 8);
+	size_sorter = roundup(sizeof(struct obj_ec_seg_sorter) * iod_nr, 8);
+	/* for oer_tgt_recx_nrs/_idxs */
+	size_tgt_nr = roundup(sizeof(uint32_t) *
+			      (oca->u.ec.e_k + oca->u.ec.e_p), 8);
+	buf_size = size_iod + size_sgl + size_oiod + size_recx + size_sorter +
+		   size_tgt_nr * iod_nr * 2 + OBJ_TGT_BITMAP_LEN;
+	D_ALLOC(buf, buf_size);
+	if (buf == NULL)
+		return -DER_NOMEM;
+
+	tmp_ptr = buf;
+	reasb_req->orr_iods = (void *)tmp_ptr;
+	tmp_ptr += size_iod;
+	reasb_req->orr_sgls = (void *)tmp_ptr;
+	tmp_ptr += size_sgl;
+	reasb_req->orr_oiods = (void *)tmp_ptr;
+	tmp_ptr += size_oiod;
+	reasb_req->orr_recxs = (void *)tmp_ptr;
+	tmp_ptr += size_recx;
+	reasb_req->orr_sorters = (void *)tmp_ptr;
+	tmp_ptr += size_sorter;
+	reasb_req->tgt_bitmap = (void *)tmp_ptr;
+	tmp_ptr += OBJ_TGT_BITMAP_LEN;
```

Comment:
Hmm, are you trying to serialization here?  I thought this is RPC packing's job? or there are other intention here, Sorry I may miss sth.